File: src/main/java/com/mhl/mycompanybackend/service/RolesService.java
package com.mhl.mycompanybackend.service;

import com.mhl.mycompanybackend.model.PermissionName;
import com.mhl.mycompanybackend.model.Permissions;
import com.mhl.mycompanybackend.model.Roles;
import com.mhl.mycompanybackend.pojo.MessageResponse;
import com.mhl.mycompanybackend.repository.PermissionsRepository;
import com.mhl.mycompanybackend.repository.RolesRepository;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

@Service
public class RolesService {
    final
    RolesRepository rolesRepository;
    final
    PermissionsRepository permissionsRepository;

    public RolesService(RolesRepository rolesRepository, PermissionsRepository permissionsRepository) {
        this.rolesRepository = rolesRepository;
        this.permissionsRepository = permissionsRepository;
    }

    public ResponseEntity<?> findAllRoles(){
        try {
            return ResponseEntity.ok().body(rolesRepository.findAll());
        } catch (Exception e){
            return ResponseEntity.badRequest().body(new MessageResponse("Something went wrong"));
        }
    }

    public ResponseEntity<?> createRole(String roleName, String permissions) {
        Permissions permission = permissionsRepository.findByName(
                        PermissionName.valueOf(permissions))
                .orElseThrow(
                        () -> new RuntimeException("Permission not found")
                );

            Roles role = new Roles(roleName, permission);
        try {
            rolesRepository.save(role);
        } catch (Exception e){
            return ResponseEntity.badRequest().body(new MessageResponse("Role exists"));
        }
        return ResponseEntity.ok().body(new MessageResponse("Role has been created"));
    }

    public Roles findRoleByName(String roleName) {
        return rolesRepository.findRolesByRoleName(roleName).orElseThrow(() -> new RuntimeException("Role not found"));
    }

    public ResponseEntity<MessageResponse> editRole(Long id, String roleName, String permission){
        try {
            Long permissionId = rolesRepository.getPermissions(permission);
            rolesRepository.updateRole(id, roleName, permissionId);
            return ResponseEntity.ok().body(new MessageResponse("Role was updated"));
        } catch (Exception e){
            return ResponseEntity.badRequest().body(new MessageResponse("Role not found"));
        }
    }
}

File: src/main/kotlin/com/mhl/mycompanybackend/service/RolesService.kt
package com.mhl.mycompanybackend.service

import com.mhl.mycompanybackend.model.PermissionName
import com.mhl.mycompanybackend.model.Roles
import com.mhl.mycompanybackend.pojo.MessageResponse
import com.mhl.mycompanybackend.repository.PermissionsRepository
import com.mhl.mycompanybackend.repository.RolesRepository
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service

@Service
class RolesService(val rolesRepository: RolesRepository, val permissionsRepository: PermissionsRepository) {
    fun findAllRoles(): ResponseEntity<*> {
        return try {
            ResponseEntity.ok().body(rolesRepository.findAll())
        } catch (e: Exception) {
            ResponseEntity.badRequest().body(MessageResponse("Something went wrong"))
        }
    }

    fun createRole(roleName: String, permissions: String): ResponseEntity<*> {
        val permission = permissionsRepository.findByName(
                PermissionName.valueOf(permissions!!))?.orElseThrow { RuntimeException("Permission not found") }
        val role = Roles(roleName, permission)
        try {
            rolesRepository.save(role)
        } catch (e: Exception) {
            return ResponseEntity.badRequest().body(MessageResponse("Role exists"))
        }
        return ResponseEntity.ok().body(MessageResponse("Role has been created"))
    }

    fun findRoleByName(roleName: String?): Roles? {
        return rolesRepository.findRolesByRoleName(roleName)?.orElseThrow { RuntimeException("Role not found") }
    }

    fun editRoleName(roleName: String?): ResponseEntity<MessageResponse> {
        return try {
            rolesRepository.updateRoleName(roleName)
            ResponseEntity.ok().body(MessageResponse("Role was updated"))
        } catch (e: Exception) {
            ResponseEntity.badRequest().body(MessageResponse("Role not found"))
        }
    }

    fun editRolePermission(roleName: String?, permissionName: PermissionName?): ResponseEntity<MessageResponse> {
        return try {
            rolesRepository.updateRolePermission(permissionName, roleName)
            ResponseEntity.ok().body(MessageResponse("Permissions was updated"))
        } catch (e: Exception) {
            ResponseEntity.badRequest().body(MessageResponse("Role not found"))
        }
    }
}
